### PR TITLE
Update zio from 1.0.13 to 2.0.0-RC2

### PR DESCRIPTION
**Pull request by earldouglas**
**TODO:**

* [x] Fix compile errors
* [x] Fix compile warnings
* [x] Fix linting failures
* [x] Fix hanging tests
    * [x] `zio.sqs.ZioSqsSpec`
    * [x] `zio.sqs.producer.ProducerErrorSpec`
    * [x] `zio.sqs.producer.ProducerEventSpec`
    * [x] `zio.sqs.producer.ProducerSpec`
        * [x] `nextPower2 can be calculated`
        * [x] `SqsRequestEntry can be created`
        * [x] `SqsRequest can be created`
        * [x] `SqsResponseErrorEntry can be created`
        * [x] `SendMessageBatchResponse can be partitioned`
        * [x] `SendMessageBatchResponse can be partitioned and mapped`
        * [x] `buildSendMessageBatchRequest creates a new request`
        * [x] `runSendMessageBatchRequest can be executed`
        * [x] `events can be published using sendStream and return the results`
        * [x] `events can be published using sendStream and fail the task on error`
        * [x] `events can be published using produce and return the results`
        * [x] `events can be pushed using produce and fail the task on error`
        * [x] `events can be published using produceBatch and return the results`
        * [x] `events can be published using produceBatch and fail the task on error`
        * [x] `events can be published using sendSink`
        * [x] `events that published using sendSink and generate an exception on send should fail the sink`
        * [x] `events that published using sendSink and return an unrecoverable error should fail the sink on error`
        * [x] `submitted events can succeed and fail if there are unrecoverable errors`
        * [x] `submitted events can be republished if there are recoverable errors`
        * [x] `if the number of recoverable retries exceeds the limit, messages fail`
        * [x] `a SendMessageBatchRequest failed with an exception should fail`
        * [x] `several SendMessageBatchRequest failed with an exception should not stop the producer (#456)`
            * Hangs on [`p.produce(e)`](https://github.com/zio/zio-sqs/blob/896ca6d/src/test/scala/zio/sqs/producer/ProducerSpec.scala#L570)
            * ~~Possible regression of #456~~
    * [x] `zio.sqs.serialization.SerializerSpec`

### CODE REVIEW

File: src/test/scala/zio/sqs/ZioSqsSpec.scala [hunk 1]
```diff
@@ -1,39 +1,38 @@
...
 
   def spec: ZSpec[TestEnvironment, Any] =
     suite("ZioSqsSpec")(
-      testM("send messages") {
+      test("send messages") {
         val settings: SqsStreamSettings = SqsStreamSettings(stopWhenQueueEmpty = true)
 
         for {
-          messages <- gen.sample.map(_.value).run(Sink.head[Chunk[String]]).someOrFailException
+          messages <- gen.sample.map(_.map(_.value).get).run(Sink.head[Chunk[String]]).someOrFailException
```
Comment (earldouglas) [inline src/test/scala/zio/sqs/ZioSqsSpec.scala]: Is there a better alternative to `.get` here?

File: src/test/scala/zio/sqs/ZioSqsSpec.scala [hunk 2]
```diff
@@ -1,39 +1,38 @@
...
+        } yield assert(list.map(_.body.getOrElse("")))(equalTo(messages))
       },
-      testM("delete messages manually") {
+      test("delete messages manually") {
         val settings: SqsStreamSettings =
           SqsStreamSettings(stopWhenQueueEmpty = true, autoDelete = false, waitTimeSeconds = Some(1))
 
         for {
-          messages <- gen.sample.map(_.value).run(Sink.head[Chunk[String]]).someOrFailException
+          messages <- gen.sample.map(_.map(_.value).get).run(Sink.head[Chunk[String]]).someOrFailException
```
Comment (earldouglas) [inline src/test/scala/zio/sqs/ZioSqsSpec.scala]: Is there a better alternative to `.get` here?

File: src/test/scala/zio/sqs/ZioSqsSpec.scala [hunk 3]
```diff
@@ -43,11 +42,11 @@ object ZioSqsSpec extends DefaultRunnableSpec {
 
         } yield assert(list)(isEmpty)
       },
-      testM("delete messages automatically") {
+      test("delete messages automatically") {
         val settings: SqsStreamSettings = SqsStreamSettings(stopWhenQueueEmpty = true, waitTimeSeconds = Some(1))
 
         for {
-          messages <- gen.sample.map(_.value).run(Sink.head[Chunk[String]]).someOrFailException
+          messages <- gen.sample.map(_.map(_.value).get).run(Sink.head[Chunk[String]]).someOrFailException
```
Comment (earldouglas) [inline src/test/scala/zio/sqs/ZioSqsSpec.scala]: Is there a better alternative to `.get` here?

File: src/test/scala/zio/sqs/ZioSqsSpec.scala (hunk 1 above)
Comment (ghostdogpr) [reply src/test/scala/zio/sqs/ZioSqsSpec.scala]: You can do `gen.runHead.someOrFailException` directly.

**Comment by ghostdogpr** [comment]
Created a `series/2.x` branch and changed the target of this PR, to be consistent with the other libraries in the ZIO ecosystem. We can publish a RC from the branch when this is ready.

**Comment by ghostdogpr** [comment]
@earldouglas Thanks! I just enabled publishing for the `series/2.x` branch, so the CI is now running and it should publish a snapshot pretty soon. It will be there: https://oss.sonatype.org/content/repositories/snapshots/dev/zio/zio-sqs_2.13/

**Comment by ghostdogpr** [comment]
Here it is: https://oss.sonatype.org/content/repositories/snapshots/dev/zio/zio-sqs_2.13/0.4.3+2-01350308-SNAPSHOT/

**Pull request merged**